### PR TITLE
romio: Gpfs large count fixup

### PR DESCRIPTION
**Pull request by roblatham00**
## Pull Request Description

Recent 64 bit "large count" work missed one spot in GPFS 
## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c [hunk 1]
```diff
@@ -682,7 +682,7 @@ void ADIOI_GPFS_Calc_others_req(ADIO_File fd, MPI_Count count_my_req_procs,
      * Receive 1 int from each process.  count_others_req_per_proc[i] is the number of
```
Comment (raffenet) [inline src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c:682]:
```suggestion
     * Receive 1 `MPI_COUNT` from each process.  count_others_req_per_proc[i] is the number of
```

**Review by raffenet — COMMENTED** [review]
LGTM, one minor nit in a comment.

File: src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c (hunk 1 above)
Comment (roblatham00) [reply src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c]: good eye.  fixed in latest revision

File: src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c [hunk 2]
```diff
@@ -677,12 +677,14 @@ void ADIOI_GPFS_Calc_others_req(ADIO_File fd, MPI_Count count_my_req_procs,
...
-    /* Send 1 int to each process.  count_my_req_per_proc[i] is the number of
+    /* Preliminary "how much work will we do?" data exchange:
+     * count_my_req_per_proc[i] is the number of
      * requests that my process will do to the file domain owned by process[i].
-     * Receive 1 int from each process.  count_others_req_per_proc[i] is the number of
+     * count_others_req_per_proc[i] is the number of
      * requests that process[i] will do to the file domain owned by my process.
-     */
-    count_others_req_per_proc = ADIOI_Malloc(nprocs * sizeof(int));
+     * Just sending/recieving one value here: a few lines later we'll set up
```
Comment (raffenet) [inline src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c:685]:
```suggestion
     * Just sending/receiving one value here: a few lines later we'll set up
```

Comment (roblatham00) [reply src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c]: i never get that right... glad the spell-checker is working

**Comment by roblatham00** [comment]
Tested this on improv.lcrc.anl.gov and confirmed the observed memory overrun and confirmed this fixed it.

File: src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c [hunk 3]
```diff
@@ -462,7 +462,7 @@ void ADIOI_GPFS_Calc_my_req(ADIO_File fd, ADIO_Offset * offset_list, ADIO_Offset
 #ifdef AGGREGATION_PROFILE
     MPE_Log_event(5024, 0, NULL);
 #endif
-    *count_my_req_per_proc_ptr = ADIOI_Calloc(nprocs, sizeof(int));
+    *count_my_req_per_proc_ptr = ADIOI_Calloc(nprocs, sizeof(**count_my_req_per_proc_ptr));
```
Comment (hzhou) [inline src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c:465]: Not sure the double dereference is pleasent on the eye, but I'll let it slide ;)

File: src/mpi/romio/test/coll_perf.c [hunk 4]
```diff
@@ -63,12 +64,12 @@ int main(int argc, char **argv)
         len = strlen(*argv);
         filename = (char *) malloc(len + 1);
         strcpy(filename, *argv);
-        MPI_Bcast(&len, 1, MPI_INT, 0, MPI_COMM_WORLD);
-        MPI_Bcast(filename, len + 1, MPI_CHAR, 0, MPI_COMM_WORLD);
+        MPI_Bcast(&len, 1, MPI_AINT, 0, MPI_COMM_WORLD);
+        MPI_Bcast_c(filename, len + 1, MPI_CHAR, 0, MPI_COMM_WORLD);
```
Comment (hzhou) [inline src/mpi/romio/test/coll_perf.c:68]: This means that we can't run the test on MPIs that does not support large count yet. Is this change necessary?

Comment (roblatham00) [reply src/mpi/romio/test/coll_perf.c:68]: no, not necessary... Instead of promoting `len` I can just cast to int -- no one is making a 2 GiB long filename

Comment (hzhou) [reply src/mpi/romio/test/coll_perf.c:68]: I see. How about `assert` before cast?
